src/python/tests/core/fuzzing/fuzzer_selection_test.py

```diff
@@ -123,6 +129,92 @@ def test_mapping_removed(self):
     self.assertEqual([], mappings)
 
 
+@test_utils.with_cloud_emulators('datastore')
+class UpdateMappingsForJobTest(unittest.TestCase):
+  """Tests for the update_mappings_for_job function."""
+
+  def setUp(self):
+    self.fuzzer_1 = data_types.Fuzzer()
+    self.fuzzer_1.name = 'fuzzer_1'
+    self.fuzzer_1.put()
+
+    self.fuzzer_2 = data_types.Fuzzer()
+    self.fuzzer_2.name = 'fuzzer_2'
+    self.fuzzer_2.put()
+
+  def test_no_mappings(self):
+    """Ensure that we do nothing if a job has no mappings."""
+    job = data_types.Job()
+    job.name = 'no_mappings'
+    job.put()
+
+    fuzzer_selection.update_mappings_for_job(job, [])
+
+    self.assertEqual(_get_fuzzer_list_for_job(job), [])
+
+  def test_new_addition(self):
+    """Ensure that we add mappings for a new job."""
+    job = data_types.Job()
+    job.name = 'new_addition'
+    job.put()
+
+    fuzzer_selection.update_mappings_for_job(job, ['fuzzer_1', 'fuzzer_2'])
+
+    mappings = _get_fuzzer_list_for_job(job)
+    self.assertIn('fuzzer_1', mappings)
+    self.assertIn('fuzzer_2', mappings)
+
+  def test_mapping_added(self):
+    """Ensure that we properly add mappings for existing jobs."""
+    job = data_types.Job()
+    job.name = 'adding_fuzzers'
+    job.put()
+
+    fuzzer_selection.update_mappings_for_job(job, ['fuzzer_1'])
+
+    mappings = _get_fuzzer_list_for_job(job)
+    self.assertIn('fuzzer_1', mappings)
+    self.assertNotIn('fuzzer_2', mappings)
+
+    fuzzer_selection.update_mappings_for_job(job, ['fuzzer_1', 'fuzzer_2'])
+
+    mappings = _get_fuzzer_list_for_job(job)
+    self.assertIn('fuzzer_1', mappings)
+    self.assertIn('fuzzer_2', mappings)
+
+  def test_mapping_subsituted(self):
```

Comment:
nit: fix spelling of subsituted. Should be "substituted".

Reply:
done 👍